bin/ui.rb

```diff
@@ -0,0 +1,104 @@
+module UI
+puts "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+puts "X   WELCOME TO TIC TAC TOE!   X"
+puts "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+
+WIN_COMBINATIONS = [[0,1,2], [3,4,5], [6,7,8], 
+                    [0,3,6], [1,4,7], [2,5,8], 
+                    [0,4,8], [2,4,6]]          
+
+def add_player(player)
+  
+    if player == 1
+    puts "Please, enter first player name"
+    $name1 = gets.chomp
+    puts "Please, enter second player name"
+    $name2 = gets.chomp
+    puts "#{$name1}, which piece do you want? (X/O)" 
+
+      piece_set = false
+      while !piece_set
+        piece = gets.chomp.upcase
+        unless piece == "X" || piece == "O"
+          puts "Please enter X or O!"
+        else
+          piece_set = true
+        end
+      end
+      @players[1] = Player.new(piece == "X" ? 1 : 2)
+    elsif player == 2
+      @players[2] = Player.new(@players[1].type == 1 ? 2 : 1)
+    end
+  end
+
+  def player_move(player)
```

Comment:
Should this method not be part of  Player class or similar

Reply:
Please, try to play here https://repl.it/@ojohnnyss/test1 and in console it works, I already gave screen to Adnan

Reply:
I collect all puts, gets in bin folder as Adnan asked

Reply:
I changed the structure of the project. Now game launch from folder bin